### PR TITLE
Improve Jenkins API usage

### PR DESCRIPTION
**Pull request by Fox32**
Avoid loading data from Jenkins twice. Don't load data when navigating thought the pages as all data from all pages is already loaded.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)

### CODE REVIEW

File: .changeset/brave-zoos-fail.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/plugin-jenkins': patch
+---
+
+Avoid loading data from Jenkins twice. Don't load data when navigating thought the pages as all data from all pages is already loaded.
```
Comment (adamdmharvey) [inline .changeset/brave-zoos-fail.md:5]:
```suggestion
Avoid loading data from Jenkins twice. Don't load data when navigating through the pages as all data from all pages is already loaded.
```

**Review by adamdmharvey — COMMENTED** [review]
Super excited by this one; thanks @Fox32 🤩

**Comment by Fox32** [comment]
> Super excited by this one; thanks @Fox32 🤩

Sadly this is only a very small improvement. It's still.loafing a lot of data. But i may also take a look into that

**Review by Rugvip — APPROVED** [review]
Yup, thanks! :grin:

**Pull request merged**